File: resource-depot/src/types/export.ts
// === MB→FBXエクスポート型定義 ===

export type ExportFormat = 'fbx' | 'glb' | 'gltf';

export type ExportStatus = 'pending' | 'running' | 'completed' | 'failed';

/** MB→FBXエクスポート設定 */
export interface MbExportConfig {
  maya_path?: string;
  output_format: ExportFormat;
  fbx_version: string;
  bake_animation: boolean;
  export_skeleton: boolean;
}

/** エクスポートジョブ */
export interface ExportJob {
  id: string;
  input_path: string;
  output_path: string;
  config: MbExportConfig;
  status: ExportStatus;
  error?: string;
  resource_id?: string;
}

File: resource-depot/src-tauri/src/models/export.rs
use serde::{Deserialize, Serialize};

/// エクスポートフォーマット
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum ExportFormat {
    Fbx,
    Glb,
    Gltf,
}

/// エクスポートの状態
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum ExportStatus {
    Pending,
    Running,
    Completed,
    Failed,
}

/// MB→FBXエクスポート設定
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MbExportConfig {
    /// Mayaの実行パス（mayapy / mayabatch）
    pub maya_path: Option<String>,
    /// 出力フォーマット
    pub output_format: ExportFormat,
    /// FBXバージョン (例: "FBX202000")
    pub fbx_version: String,
    /// アニメーションをベイクするか
    pub bake_animation: bool,
    /// スケルトンをエクスポートするか
    pub export_skeleton: bool,
}

impl Default for MbExportConfig {
    fn default() -> Self {
        Self {
            maya_path: None,
            output_format: ExportFormat::Fbx,
            fbx_version: "FBX202000".to_string(),
            bake_animation: true,
            export_skeleton: true,
        }
    }
}

/// エクスポートジョブ
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ExportJob {
    pub id: String,
    /// 入力ファイルパス (.mb)
    pub input_path: String,
    /// 出力ファイルパス (.fbx)
    pub output_path: String,
    /// エクスポート設定
    pub config: MbExportConfig,
    /// 状態
    pub status: ExportStatus,
    /// エラーメッセージ
    pub error: Option<String>,
    /// 対応するリソースID（登録済みの場合）
    pub resource_id: Option<String>,
}
